.github/workflows/clean_validation_envs.yml

```diff
@@ -0,0 +1,71 @@
+---
+name: Clean Validation Environments
+
+on:
+  schedule:
+    # 2am each night https://crontab.guru/#0_2_*_*_*
+    - cron: "0 2 * * *"
+  workflow_dispatch:
+
+jobs:
+  clean:
+    name: Clean
+    runs-on: ubuntu-latest
+    environment: Dev
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # This is CRITICAL since we're making decisions based on branch existence
+          fetch-depth: 0
+
+      - name: Azure Login
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: clean
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -o errexit
```

Comment:
I think this is great, I could see this having a benefit in AzDO builds too. If this was in a script, behind something like `make deleteEnvironment` it would allow us to call it locally and also means we  have less logic in GH Actions

Reply:
I understand moving the logic to a script file, but not sure about putting it inside the make file - it doesn't really have anything to do with the tre solution but more as a tiding step of the ci.
Will it make sense to do it without the make target and call the script directly in the ci step?

Reply:
Done @ross-p-smith